build-logic/build.gradle.kts

```diff
@@ -17,6 +24,39 @@ kotlin {
             .get()
             .toInt(),
     )
+
+    compilerOptions {
+        // Match language version enforced by current Gradle version https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+        @Suppress("DEPRECATION")
+        apiVersion.set(KotlinVersion.KOTLIN_1_8)
+    }
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    compilerOptions {
+        freeCompilerArgs.add("-Xsuppress-version-warnings") // ignores deprecated kotlin language version
```

Comment:
Prevents 
> w: API version 1.8 is deprecated and its support will be removed in a future version of Kotlin

polluting the console

<img width="724" alt="image" src="https://github.com/user-attachments/assets/1c0462d5-092f-45ac-b73f-641e9f4c023c" />